wordpress-mobile/WordPress-Android
# [Site Creation] Design Review

Fixes #8864 & #8883. @SylvesterWilmott and I have gone through a design review together on a Slack call and found several small issue to address. These issues have been documented in #8864 and addressed in this PR.  Here is a list of changes made in this PR:

* Adds consistent start & end margins to each site creation page using the `content_margin` which has different values depending on the device size. We were originally supposed to add a `maxWidth` but @SylvesterWilmott mentioned that he liked this approach by @theck13 in #8860 so we are going with that instead. It's also consistent with what the rest of the app does for tablet layouts.
* We decided to handle the missing `iconUrl` issue in the backend.
* Introduced styles for the common UI components and made sure they are consistent (I haven't done an exhaustive check, but I have extracted some bits and pieces as I found them)
* Fixed the widow word issues brought up by @theck13 [here](https://github.com/wordpress-mobile/WordPress-Android/pull/8863#pullrequestreview-187747414)
* We are now auto-focusing the first `EditText` in each page, but we are not opening the keyboard in the site info page and we are only opening it in domains page if the site doesn't have a title. The idea is to not block content when the user enters the screen. If the site has a title, we'll try to show domain suggestions, so showing the keyboard might potentially block content when the request is finished.
* Added a skeleton to the shimmer animation in the site preview page.
* Custom logic for showing the `Links are disabled` toast has been removed. It turns out this was working as intended before.
* When we receive the domain suggestions we'll show the results that matches the current query at the top of the list as a quality of life improvement.
* Updated ***Skip*** button margins as described in #8883.
* Verticals screen now has the same container as site info for the skip button
* Create site button in domains screen is now 36dp and it has a rounded corner similar to skip button in site info page
* Added shadow to the site url container in the site preview page

### Design Review

As mentioned, we have already gone through a design pass, but since we have a lot of fixes, I think it's a good idea to do it again. This will also serve as a documentation for all the designs that are implemented in this project.

First of here is me going through the site creation flow for different devices and orientations:

[Phone portrait](https://cloudup.com/c2PNlq3Fo_h)
[Phone landscape](https://cloudup.com/cCNmWcpnvdy)
[Tablet landscape](https://cloudup.com/comyoMDarNb)
[Tablet portrait](https://cloudup.com/cyXRkGTaf7E) _Sorry about the quality, the gif ended up being in wrong orientation and while fixing that I accidentally lowered the quality of it_

Here are the screenshots for successful flow:

[Phone portrait](https://cloudup.com/chBk0CXnhQr)
[Phone landscape](https://cloudup.com/cCHgnvKe6Yf)
[Tablet portrait](https://cloudup.com/cl8Uqwpk6pM)
[Tablet landscape](https://cloudup.com/cEe0PTDt1tI)

Here are screenshots for the errors:

[Phone portrait](https://cloudup.com/c-p_ED7dM1Y)
[Phone landscape](https://cloudup.com/cVD8t6aUK_d)
[Tablet portrait](https://cloudup.com/cDwne1EOBT8)
[Tablet landscape](https://cloudup.com/cTMo_b0VTVt)

### To test:

Going through the site creation flow following the recorded gifs for a phone and a tablet would probably be the best way to test this PR. We'll most likely have a separate PR to test the whole flow, so I think we can just focus on the changes of this PR which are many to list as testing instructions.

@theck13 I was wondering if you could check out the screenshots or the branch as well as another set of eyes? You don't have to review the code, but I'd feel more comfortable if you have gone through the flow at least once.
